### PR TITLE
FEATURE: Let me talk to myself

### PR DESCRIPTION
**Pull request by nattsw**
And also put my own DM at the top

Chat screen showing a DM to myself and the sidebar
.

### CODE REVIEW

**Comment by nattsw** [comment]
@markvanlan I feel like you would know this better: My most recent commit was to remove the transaction I added when creating a DM. I found the transaction useful when I was dev-ing so errors along the way would be properly reverted. I don't know if we want it as a thing, so I've removed it and we can add it back if it sounds useful.

File: app/controllers/direct_messages_controller.rb [hunk 1]
```diff
@@ -5,12 +5,9 @@ def create
     guardian.ensure_can_chat!(current_user)
     params.require(:usernames)
 
-    users = User
-      .where(username: params[:usernames].split(","))
-      .to_a
-      .concat([current_user])
+    users = [current_user]
+    users += User.where(username: params[:usernames].split(",")).to_a if current_user.username != params[:usernames]
```
Comment (markvanlan) [inline app/controllers/direct_messages_controller.rb:9]: We generally prefer `<<` as it's a much faster operation

File: assets/javascripts/discourse/services/chat.js [hunk 2]
```diff
@@ -319,6 +319,14 @@ export default Service.extend({
         this.currentUser.chat_channel_tracking_state[a.id]?.unread_count || 0;
       const unreadCountB =
         this.currentUser.chat_channel_tracking_state[b.id]?.unread_count || 0;
+
+      const userAIsSelf = a.chatable.users[0].id === this.currentUser.id;
+      const userBIsSelf = b.chatable.users[0].id === this.currentUser.id;
```
Comment (markvanlan) [inline assets/javascripts/discourse/services/chat.js:324]: I don't think your personal DM channel should be at the top of the list always, other chat apps don't do this that I've seen. Was this just something you thought would be good, or have you seen this elsewhere?

Comment (nattsw) [reply assets/javascripts/discourse/services/chat.js:324]: Slack does that! 🤔

**Review by markvanlan — APPROVED** [review]

File: assets/javascripts/discourse/services/chat.js (hunk 2 above)
Comment (nattsw) [reply assets/javascripts/discourse/services/chat.js:324]: Leaving this as-is, since we talked about it in chat. :p

Comment (CvX) [reply assets/javascripts/discourse/services/chat.js:324]: I think I'll avoid messaging myself until there's the ability to close a conversation 😅 I don't want to be stuck with least relevant chat at the top! 😛

**Pull request merged**